### DISCUSSION TITLE
Using VectorPostprocessor values as boundary conditions

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#22845] [Q&A General] [answered]
Hello,

I want to use the values on a boundary on one 2D subdomain as the boundary condition of another 2D subdomain. 
I am trying to use the values from a vector postprocessor (SideValueSampler) as the boundary condition for the other domain. 
My field variable is not a vector. 

How do I go about this?

Thank you.

**Comment by GiudGiud** [comment]
Hello

Are the 2D subdomain physically connected in the mesh?

Guillaume

**Comment by salaudeen-ya** [comment]
I am assuming they are not physically connected but they have the same mesh
sizes and dimension.

My main aim is to decompose a 2D domain into subdomains and still obtain
the same result as if they are a single domain.

**Reply by GiudGiud** [reply]
so we do not have a VPPDirichletBC you could use.
And if you went through CSV output you would necessarily lag a bit. The minimum lag would be 1 linear iteration, if you run the CSV output on linear (expensive to write all the time). It's doable though if you want to try:

- output using a SideValueSampler
- read using a PropertyReadFile
- sample using a PiecewiseConstantFromCSV on a nearest-node basis OR if the ids match nice on a per-element basis
- use a function dirichlet BC

What I would advocate instead is a multiapp approach. Then you can use a nearest node transfer to match two boundaries that are close but disjoint. This will let you synchornize those values every time step using fixed point iterations.

**Comment by salaudeen-ya** [comment]
Awesome. Thanks for the tip!

So, I was actually using Multi-App. Currently I need to use the values
transferred using nearest node from the other subdomain as the BC for the
second domain.

I can see the transfer. But since the values from the nearest node is saved
in an aux variable, it's proving difficult to use this aux value as BC. Any
idea on how to do that?

Thanks.

**Comment by GiudGiud** [comment, marked as answer]
To use the dirichlet BC from a variable I think you want to look at the matchedValueBC
